solr/core/src/java/org/apache/solr/handler/admin/CoreAdminHandler.java

```diff
@@ -436,7 +436,8 @@ public static class CoreAdminAsyncTracker {
      * we're not supposed to hit it. This is just a protection to grow in memory too much when
      * receiving an abusive number of admin requests.
      */
-    private static final int MAX_TRACKED_REQUESTS = 10_000;
+    private static final int MAX_TRACKED_REQUESTS =
+        EnvUtils.getPropertyAsInteger("solr.admin.requests.running.max", 10_000);
```

Comment:
As this is only for "async" requests, maybe the name should incorporate this limit like 'solr.admin.async.max". ("requests" and "running" being kind of implicit maybe)?